src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -338,7 +339,12 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
       performanceTracking.getState().executeFn({
         fn: () => {
           const { routes, index } = Navigation.getState();
-          if (index === 0 || routes[index - 1].name === Routes.EXPANDED_ASSET_SHEET) {
+          const activeRoute = Navigation.getActiveRoute();
+          if (
+            index === 0 ||
+            routes[index - 1] === Routes.EXPANDED_ASSET_SHEET ||
+            activeRoute.name === Routes.PAIR_HARDWARE_WALLET_AGAIN_SHEET
+          ) {
```

Comment:
I know this wasn't introduced in your PR, but can we specify what these checks are? I'm a little confused what `index === 0` implies. Something like:

```
const isOnlyRoute = index === 0;  // not sure if this is actually what this means?
const isFromExpandedSheet = ...
const isFromHardwareWalletSheet = ...
```

Reply:
I have a feeling that both of those cases do not fit the typical shape of `routes`. I hadn't yet had time to check out that expanded sheet flow yet.

I'll try to verify that either of these are necessary. Then make them more legible.